float32 goalx
float32 goaly
---
float32 distance